pkg/lib/queueinformer/queueinformer_operator.go

```diff
@@ -273,8 +273,8 @@ func (o *operator) processNextWorkItem(ctx context.Context, loop *QueueInformer)
 	logger := o.logger.WithField("item", item)
 	logger.WithField("queue-length", queue.Len()).Trace("popped queue")
 
-	event, ok := item.(kubestate.ResourceEvent)
-	if !ok || event.Type() != kubestate.ResourceDeleted {
+	var event = item
+	if item.Type() != kubestate.ResourceDeleted {
 		// Get the key
```

Comment:
There's another nutty thing going on here (which I dared not touch). The queue items are in the form
EventResource{type, string} (with the exception of the delete event which carries the object), but the events that get sent to the Sync'ers are in the form EventResource{type, obj}.
So, initial intuition that the syncers worked over old/failed versions of the resources was wrong, because the current version gets picked up and injected here. 

Or, did you already get that, and there's a better way to massage this that I'm not seeing?

Reply:
the reason why the delete event doesn't get the same treatment is: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/lib/queueinformer/queueinformer.go#L40. 
We may want to revisit this in a follow-up. But, for now, I feel there is sufficient change in this PR that we can let out into the wild before further refactoring.